### ISSUE TITLE
Fallback-functionality

### ISSUE THREAD

**Issue by thomkle** [hishamco/My.Extensions.Localization.Json#25]
Any plans to support fallback-functionality? E.g. say I have a TestController and the files testcontroller.es-ES.json, testcontroller.es.json and testcontroller.json. The fallback-functionality would first check the testcontroller.es-ES.json for the key, then the testcontroller.es.json and then testcontroller.json.

Edit: Like explained [here](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/localization?view=aspnetcore-3.0#culture-fallback-behavior)

**Comment by hishamco** [comment]
Any plans to support fallback-functionality?

Sure, let me check why this not included, I already support this in some OSS

**Comment by hishamco** [comment]
FYI the fallback is here https://github.com/hishamco/My.Extensions.Localization.Json/blob/master/src/My.Extensions.Localization.Json/JsonStringLocalizer.cs#L119-L140

Are you sure it doesn't work, perhaps I need to add a unit test if it is not included yet

**Comment by thomkle** [comment]
That's really strange, to provide more accurate info: 
I have a test.en-US.json, test.en.json and test.json (in the same folder) with the following content:

test.en-US.json
```
{
  "temp":  "test-en-US"
}
```

test.en.json
```
{
  "temp": "test-en",
  "temp2": "test-en"
}
```
test.json
```
{
  "temp": "test",
  "temp2": "test",
  "temp3": "test"
}
```

With an empty Test-class:
```
public class Test
{
}
```

And then trying to access the values using `IStringLocalizer<Test> _localizer` (injected into a controller) and `_localizer["temp"] == "test-en-US"` (which is correct), but `_localizer["temp2"] == "temp2"` and `_localizer["temp3"] == "temp3"`.

Do you have an example where the fallback-functionality is used? Maybe I'm missing something.

**Comment by hishamco** [comment]
I know what you mean exactly, let me create a unit test for that

**Comment by thomkle** [comment]
Great, thanks for being so fast to respond!

**Comment by hishamco** [comment]
@thomkle seems the culture fallback works fine with `GetAllStrings()` but not localizer indexer

**Comment by thomkle** [comment]
@hishamco Ah, I see. I'll follow the status on #26

**Comment by hishamco** [comment]
It's already merged ;) feel free to use and try the latest bits

**Comment by thomkle** [comment]
Ah, great! I'll try out the last change. Could you push a NuGet package for it as well?

**Comment by thomkle** [comment]
@hishamco Seems to be working very well now, thanks!

**Comment by thomkle** [comment]
@hishamco Sorry to nag, but the fallback doesn't seem to work completely. E.g. say I have set the culture "es-ES", but there are no files <filename>.es-ES.json, but I have a <filename>.json. It will not fall back to values in <filename>.json. I also tested this creating unit tests after cloning the repo and it looks that the unit test fails. 
Reproduction steps:
- Add a `Test.json` with the content `{
  "Hello": "Hello there"
}`
- In the `JsonStringLocalizerTests` for the method `GetTranslation` add the inline data: `[InlineData("es-ES", "Hello", "Hello there")]`. This will fail, but this kind of fallback should work, right?

I'll be happy to try to solve this my self and create pull requests, but I understand if you're more comfortable with solving things in the project yourself :)
